fabric-network/test/impl/event/transactioneventhandler.js

```diff
@@ -6,241 +6,276 @@
-			stubEventHub._onEventFn = onEventFn;
-			stubEventHub._onErrorFn = onErrorFn;
-		});
+	let handler;
 
-		stubStrategy = {
-			getEventHubs: sinon.stub(),
-			eventReceived: sinon.stub(),
-			errorReceived: sinon.stub()
+	beforeEach(() => {
+		revert = [];
+		sandbox = sinon.createSandbox();
+		FakeLogger = {
+			debug: () => {
+			},
+			error: () => {
+			},
+			warn: () => {
+			}
 		};
-		stubStrategy.getEventHubs.returns([stubEventHub]);
+		sandbox.stub(FakeLogger);
+		revert.push(TransactionEventHandler.__set__('logger', FakeLogger));
+		eventServiceManager = sinon.createStubInstance(EventServiceManager);
+		eventServiceManager.startEventService.resolves();
+		network = sinon.createStubInstance(Network);
+		network.eventServiceManager = eventServiceManager;
+
+		transaction = sinon.createStubInstance(Transaction);
+		transaction.transactionId = transactionId;
+		contract = sinon.createStubInstance(Contract);
+		contract.network = network;
+		transaction.contract = contract;
+		transaction.getNetwork.returns(network);
+
+		eventService = sinon.createStubInstance(EventService);
+		eventService.name = 'eventService';
+		eventService.unregisterEventListener = sinon.stub();
+		baseListener = sinon.stub();
+		baseListener.eventService = eventService;
+		eventService.registerTransactionListener.returns(baseListener);
+
+		strategy = sinon.createStubInstance(EventStrategy);
+		strategy.getEventServices.returns([eventService]);
+
+		handler = new TransactionEventHandler(transaction, strategy, options);
 	});
 
+
 	afterEach(() => {
+		if (revert.length) {
+			revert.forEach(Function.prototype.call, Function.prototype.call);
+		}
+		sandbox.restore();
 		sinon.restore();
 	});
 
 	describe('#constructor', () => {
-		it('has a default timeout of zero if no options supplied', () => {
-			const handler = new TransactionEventHandler(stubTransaction, stubStrategy);
-			expect(handler.options.commitTimeout).to.equal(0);
+		it('has a default timeout', () => {
+			const test = new TransactionEventHandler(transaction, strategy);
+			expect(test.options.commitTimeout).to.equal(30);
+			expect(test.eventServices.length).to.be.equal(1);
+			expect(test.transactionId).to.be.equal(transactionId);
 		});
 
 		it('uses timeout from supplied options', () => {
-			const options = {commitTimeout: 1};
-			const handler = new TransactionEventHandler(stubTransaction, stubStrategy, options);
-			expect(handler.options.commitTimeout).to.equal(options.commitTimeout);
-		});
+			const toptions = {commitTimeout: 1};
 
-		it('should set transactionID and transaction', () => {
-			const handler = new TransactionEventHandler(stubTransaction, stubStrategy);
-			expect(handler.transactionId).to.equal(transactionId);
-			expect(handler.transaction).to.equal(stubTransaction);
+			const test = new TransactionEventHandler(transaction, strategy, toptions);
+			expect(test.options.commitTimeout).to.equal(toptions.commitTimeout);
 		});
 	});
 
-	describe('event handling:', () => {
```

Comment:
The previous tests provided a functional spec of handler behaviour and should be directly applicable to any new implementation since the handler behaviour needs to remain consistent. The new tests seem to test specific elements of internal behaviour without the context of what this means to code that uses the handler.

Reply:
fixed